### PR TITLE
D4J-39 - Added metod's declaration

### PR DESCRIPTION
**Pull request by denisDomashenko**

### CODE REVIEW

**Comment by IgorKey** [comment]
Давай тесты. Тут они прям идеально выглядят

File: plugins/drill-coverage-plugin/src/adminPartMain/kotlin/JacocoExt.kt [hunk 1]
```diff
@@ -45,3 +45,29 @@ fun ICoverageNode.coverageKey(parent: ICoverageNode? = null): CoverageKey = when
...
+            "B" -> declList.add("; byte")
+            "D" -> declList.add("; double")
+            "S" -> declList.add("; short")
+            "C" -> declList.add("; char")
+            else -> declList.add("; ${arg.substringAfterLast('/')}")
+        }
+    }
+    var decl = declList.joinToString(prefix = "(")
+    decl = "${decl.substringBeforeLast("; ")}) : ${decl.substringAfterLast("; ")}".replace("(; ", "(").replace(", ", "")
+    println(decl)
```
[the remark this answers is not in the document]
Comment (denisDomashenko) [reply plugins/drill-coverage-plugin/src/adminPartMain/kotlin/JacocoExt.kt]: done

File: plugins/drill-coverage-plugin/src/adminPartMain/kotlin/JacocoExt.kt [hunk 2]
```diff
@@ -45,3 +45,29 @@ fun ICoverageNode.coverageKey(parent: ICoverageNode? = null): CoverageKey = when
...
+
+fun CoverageKey.declaration(desc: String): String {
+    val argsString = desc.substringAfter('(').replace(")", "")
+    val declList = mutableListOf<String>()
+    val args = if (argsString.endsWith(';')) argsString.substringBeforeLast(';').split(';')
+    else argsString.split(';')
+
+    args.forEach {arg ->
+        when (arg) {
+            "V" -> declList.add("; void")
```
[the remark this answers is not in the document]
Comment (denisDomashenko) [reply plugins/drill-coverage-plugin/src/adminPartMain/kotlin/JacocoExt.kt]: Done

File: plugins/drill-coverage-plugin/src/adminPartMain/kotlin/JacocoExt.kt [hunk 3]
```diff
@@ -45,3 +45,29 @@ fun ICoverageNode.coverageKey(parent: ICoverageNode? = null): CoverageKey = when
...
+            "I" -> declList.add("; int")
+            "F" -> declList.add("; float")
+            "B" -> declList.add("; byte")
+            "D" -> declList.add("; double")
+            "S" -> declList.add("; short")
+            "C" -> declList.add("; char")
+            else -> declList.add("; ${arg.substringAfterLast('/')}")
+        }
+    }
+    var decl = declList.joinToString(prefix = "(")
```
[the remark this answers is not in the document]
Comment (denisDomashenko) [reply plugins/drill-coverage-plugin/src/adminPartMain/kotlin/JacocoExt.kt]: Done

File: plugins/drill-coverage-plugin/src/adminPartMain/kotlin/JacocoExt.kt [hunk 4]
```diff
@@ -45,3 +45,29 @@ fun ICoverageNode.coverageKey(parent: ICoverageNode? = null): CoverageKey = when
...
+    decl = "${decl.substringBeforeLast("; ")}) : ${decl.substringAfterLast("; ")}".replace("(; ", "(").replace(", ", "")
```
[the remark this answers is not in the document]
Comment (denisDomashenko) [reply plugins/drill-coverage-plugin/src/adminPartMain/kotlin/JacocoExt.kt]: Done

File: plugins/drill-coverage-plugin/src/adminPartMain/kotlin/JacocoExt.kt [hunk 5]
```diff
@@ -45,3 +45,31 @@ fun ICoverageNode.coverageKey(parent: ICoverageNode? = null): CoverageKey = when
...
     else -> CoverageKey(this.name.crc64)
 }
+
+fun CoverageKey.declaration(desc: String): String {
+    val argsString = desc.substringAfter('(').replace(")", "")
+    val declList = mutableListOf<String>()
+    val args = if (argsString.endsWith(';')) argsString.substringBeforeLast(';').split(';')
+    else argsString.split(';')
+
+    args.forEach {arg ->
```
Comment (viktorbll) [inline plugins/drill-coverage-plugin/src/adminPartMain/kotlin/JacocoExt.kt]: This should be replaced by map function.

File: plugins/drill-coverage-plugin/src/adminPartMain/kotlin/JacocoExt.kt [hunk 6]
```diff
@@ -45,3 +45,31 @@ fun ICoverageNode.coverageKey(parent: ICoverageNode? = null): CoverageKey = when
...
+            "B" -> declList.add("byte")
+            "D" -> declList.add("double")
+            "S" -> declList.add("short")
+            "C" -> declList.add("char")
+            else -> declList.add("${arg.substringAfterLast('/')}")
+        }
+    }
+    var decl = declList.joinToString(separator = "; ")
+    decl =
+        if (decl.contains(";")) "(${decl.substringBeforeLast("; ")}) : ${decl.substringAfterLast("; ")}"
```
Comment (viktorbll) [inline plugins/drill-coverage-plugin/src/adminPartMain/kotlin/JacocoExt.kt]: This is confusing and not needed. "joinToString" has prefix and suffix params.

**Pull request closed without merging**